forge-gui/res/cardsfolder/upcoming/regenerations_restored.txt

```diff
@@ -0,0 +1,11 @@
+Name:Regenerations Restored
+ManaCost:W U
+Types:Enchantment
+K:Vanishing:12
+T:Mode$ CounterRemovedOnce | ValidCard$ Card.Self | CounterType$ TIME | TriggerZones$ Battlefield | Execute$ TrigScry | TriggerDescription$ Whenever one or more time counters are removed from CARDNAME, scry 1 and you gain 1 life. Then if CARDNAME has no time counters on it, exile it. When you do, take an extra turn after this one.
+SVar:TrigScry:DB$ Scry | ScryNum$ 1 | SubAbility$ DBGainLife
+SVar:DBGainLife:DB$ GainLife | LifeAmount$ 1 | SubAbility$ DBExile
+SVar:DBExile:DB$ ChangeZone | Origin$ Battlefield | Destination$ Exile | Defined$ ReplacedCard | ConditionPresent$ Card.Self+counters_GE0_TIME | SubAbility$ DBExtraTurn
+SVar:DBExtraTurn:DB$ AddTurn | Defined$ You | NumTurns$ 1 | ConditionPresent$ Card.Self+counters_GE0_TIME | SpellDescription$ Take an extra turn after this one.

```

Comment:
Is there a more efficient way to write this without including the condition twice?

Reply:
ReplacedCard doesn't make sense outside of replacement effects
"when you do" is always reflexive trigger

Reply:
In other cases you could use BranchEffect but in this case (as TRT mentioned) you'd use ImmediateTrigger

Reply:
I updated the script last night to use an ImmediateTrigger. However, I'm running into an issue where it gets sacrificed to Vanishing before the exile effect is able to take effect. Reordering the effects manually works, but if you just click "auto" you don't get the extra turn. Is there any way to get it to stay true to the rules as written while making it so that it prioritizes exiling it before sacrificing it by default?

(I'll be able to get the updated script loaded on here in an hour or so.)

Reply:
no, that's how it works by the rules

Reply:
Are you saying that what I've described isn't how it should work, or that it's exactly how it should work?

Reply:
Both CounterRemoved* events here happen while you're resolving the upkeep trigger and then
> 603.3b. If multiple abilities have triggered since the last time a player received priority, the abilities are placed on the stack in a two-part process. First, each player, in APNAP order, puts each triggered ability they control with a trigger condition that isn't another ability triggering on the stack in any order they choose.

I think the GUI supports remembering of the last order though

When you choose to sacrifice it first you will have a new object in your graveyard that the remaining trigger from the battlefield version is by no means directly involved with and therefore the event for the reflexive trigger doesn't happen
It requires explicit exceptions when that is supposed to work (see 400.7)